petstore/src/main/scala/io/finch/petstore/endpoint.scala

```diff
@@ -0,0 +1,227 @@
+package io.finch.petstore
+
+import com.twitter.finagle.Service
+import com.twitter.finagle.httpx.{Request, Response}
+import com.twitter.util.{Future}
+import io.finch.argonaut._
+import io.finch.request._
+import io.finch.response._
+import io.finch.route._
+
+/**
+ * Provides the paths and endpoints for all the API's public service methods.
+ */
+object endpoint extends ErrorHandling {
+
+  /**
+   * Private method that compiles all pet service endpoints.
+   * @return Bundled compilation of all pet service endpoints.
+   */
+  private def petEndpts(db: PetstoreDb) = getPetEndpt(db) :+: addPetEndpt(db) :+: updatePetEndpt(db) :+:
+      getPetsByStatusEndpt(db) :+: findPetsByTagEndpt(db) :+: deletePetEndpt(db) :+: updatePetViaFormEndpt(db) :+:
+      uploadImageEndpt(db)
+
+  /**
+   * Private method that compiles all store service endpoints.
+   * @return Bundled compilation of all store service endpoints.
+   */
+  private def storeEndpts(db: PetstoreDb) = getInventoryEndpt(db) :+: addOrderEndpt(db) :+: deleteOrderEndpt(db) :+:
+      findOrderEndpt(db)
+
+  /**
+   * Private method that compiles all user service endpoints.
+   * @return Bundled compilation of all user service endpoints.
+   */
+  private def userEndpts(db: PetstoreDb) = addUserEndpt(db) :+: addUsersViaList(db) :+: addUsersViaArray(db) :+:
+      getUserEndpt(db) :+: deleteUserEndpt(db) :+: updateUserEndpt(db)
+
+  /**
+   * Compiles together all the endpoints relating to public service methods.
+   * @return A service that contains all provided endpoints of the API.
+   */
+  def makeService(db: PetstoreDb): Service[Request, Response] = handleExceptions andThen (
+    petEndpts(db) :+:
+    storeEndpts(db) :+:
+    userEndpts(db)
+  ).toService
+
+  /**
+   * The long passed in the path becomes the ID of the Pet fetched.
+   * @return A Router that contains the Pet fetched.
+   */
+  def getPetEndpt(db: PetstoreDb): Router[Future[Pet]] = Get / "pet" / long /> {petId: Long =>
+    db.getPet(petId)
+  }
+
+  /**
+   * The pet to be added must be passed in the body.
+   * @return A Router that contains a RequestReader of the ID of the Pet added.
+   */
+  def addPetEndpt(db: PetstoreDb): Router[RequestReader[Long]] = Post / "pet" />{
+    body.as[Pet].embedFlatMap {pet =>
+      val p: Future[Long] = db.addPet(pet)
+      for{
+        realId <- p
+      }yield realId
+    }
+  }
+
+  /**
+   * The updated, better version of the current pet must be passed in the body.
+   * @return A Router that contains a RequestReader of the updated Pet.
+   */
+  def updatePetEndpt(db: PetstoreDb): Router[RequestReader[Pet]] = Put / "pet" /> {
+    val rr: RequestReader[Pet] = body.as[Pet].embedFlatMap { pet =>
+      val f: Future[Pet] = for {
```

Comment:
There's not really any need for a new `val` here (or just above) since we immediately return it.